### PR TITLE
Fixed sample

### PR DESCRIPTION
**Pull request by kant2002**

### CODE REVIEW

File: Samples/TdLib.Samples.GetChats/Program.cs [hunk 1]
```diff
@@ -3,8 +3,9 @@
 using System.Threading;
 using System.Threading.Tasks;
 using TDLib.Bindings;
+using TdLib;
 
-namespace TdLib.Samples.GetChats;
```
Comment (ForNeVeR) [inline Samples/TdLib.Samples.GetChats/Program.cs:7]: Well, technically, the project is called `TDLib.Samples.GetChats` (with capital `D`), and I expect the namespace to be called the same.

But let's wait for #114 before settling on this.

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

**Comment by x2bool** [comment]
Hi! Could you provide more info about what are you trying to fix? I thought we broke the samples project somehow, but I checked, and it works just fine.

**Comment by kant2002** [comment]
For context, I never use that library and even wrote Telegram bots. So I copy sample, change namespace and things blow up because `TdClient` in the `TdLib` namespace. I think that's bad experience for novices. Also I notice other things which are not clear for people new to Telegram bot development. Without ApplicationVersion samples refuses to work.

**Comment by ForNeVeR** [comment]
Could you please rebase on `master`?

**Comment by kant2002** [comment]
Please take a look. Also notice that I have to add extension methods.
I keep samples based on the 1.8.1 and not on the master version of code, since that's version which people will use.

File: Samples/TdLib.Samples.GetChats/Program.cs [hunk 2]
```diff
@@ -20,6 +25,12 @@ internal static class Program
 
     private static async Task Main()
     {
+        // TdLib creates database in the current directory.
+        // Creating separate directory and switch to that dir.
+        var clientDb = Path.Combine(AppContext.BaseDirectory, "db");
+        Directory.CreateDirectory(clientDb);
+        Directory.SetCurrentDirectory(clientDb);
```
Comment (x2bool) [inline Samples/TdLib.Samples.GetChats/Program.cs:32]: A better solution: just provide `DatabaseDirectory` and `FilesDirectory` config parameters when SetTdlibParameters called.

Comment (kant2002) [reply Samples/TdLib.Samples.GetChats/Program.cs:32]: That’s for new version. Samples which works only from master branch is disservice to the new users. That require knowledge of the changes. I’m not sure about your release cycle so wanna play safe

Comment (x2bool) [reply Samples/TdLib.Samples.GetChats/Program.cs:32]: hmm... I think there is a misunderstanding between us. DatabaseDirectory and FilesDirectory params already exist, not only in master but also in NuGet versions (even older versions). We just need to provide them here: https://github.com/egramtel/tdsharp/blob/master/Samples/TdLib.Samples.GetChats/Program.cs#L101

Comment (kant2002) [reply Samples/TdLib.Samples.GetChats/Program.cs:32]: Oops. My bad. Let me update sample then

Comment (kant2002) [reply Samples/TdLib.Samples.GetChats/Program.cs]: Done.

File: Samples/TdLib.Samples.GetChats/Program.cs [hunk 3]
```diff
@@ -1,16 +1,21 @@
 ﻿using System;
+using System.IO;
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
-using TdLib.Bindings;
```
Comment (ForNeVeR) [inline Samples/TdLib.Samples.GetChats/Program.cs:5]: Please revert this back, and you won't need the new extensions in `LegacyCompat`.

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

File: Samples/TdLib.Samples.GetChats/Program.cs (hunk 3 above)
Comment (kant2002) [reply Samples/TdLib.Samples.GetChats/Program.cs:5]: Okay. I'm prepared to discuss this.
If I revert this code, then sample would not be valid for `TDSharp=1.8.1`. If I use `using TDLib.Bindings;` then I need compat.

That's sample which working with `TDSharp=1.8.1` https://github.com/kant2002/tdsharp-samples

Comment (ForNeVeR) [reply Samples/TdLib.Samples.GetChats/Program.cs:5]: I don't think we need code valid for 1.8.1 in the `master` branch.

Moreover: we don't need code in `LegacyCompat` which will "restore" the compatibility with your sample which is in `master` but tries to use 1.8.1 APIs for whatever reason.

Just drop the legacy stuff.

Comment (kant2002) [reply Samples/TdLib.Samples.GetChats/Program.cs:5]: Last question. What's proper location for samples which works on the latest released version?

Comment (ForNeVeR) [reply Samples/TdLib.Samples.GetChats/Program.cs:5]: I'd guess they should just be in the repository.

If someone wants a sample which works for an older release, they're free to checkout the older release tag and take the sample from there.

Comment (kant2002) [reply Samples/TdLib.Samples.GetChats/Program.cs:5]: I think I'm done with that sample

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**